frontend/src/metabase/dashboard/selectors.ts

```diff
@@ -320,6 +320,19 @@ export const getParameters = createSelector(
   },
 );
 
+export const getRequiredParameters = createSelector(
+  [getParameters],
+  parameters => parameters.filter(param => param.required),
+);
+
+export const getMissingRequiredParameters = createSelector(
+  [getRequiredParameters],
+  requiredParameters =>
+    requiredParameters.filter(
+      param => !param.default || param?.default.length === 0,
```

Comment:
1. Redundant `?`
2. Let's make sure it's an array if we're inspecting `length`

```suggestion
      param => !param.default || (Array.isArray(param.default) && param.default.length === 0),
```